plugins/mobile-plugin/src/main/java/ru/sbtqa/tag/pagefactory/mobile/utils/SwipeUtils.java

```diff
@@ -147,14 +153,14 @@ public static void swipeToText(DirectionStrategy direction, String text, MatchSt
      */
     public static void swipeToText(DirectionStrategy direction, String text, MatchStrategy strategy, int depth) throws SwipeException {
         for (int depthCounter = 0; depthCounter < depth; depthCounter++) {
-            String oldPageSource = PageFactory.getDriver().getPageSource();
+            String oldPageSource = appiumDriver.getPageSource();
             switch (strategy) {
                 case EXACT:
-                    if (PageFactory.getMobileDriver().findElementsByXPath("//*[@text='" + text + "']").size() > 0) {
+                    if (appiumDriver.findElementsByXPath("//*[@text='" + text + "']").size() > 0) {
                         return;
                     }
                 case CONTAINS:
-                    List<WebElement> textViews = PageFactory.getMobileDriver().findElementsByClassName("android.widget.TextView");
+                    List<WebElement> textViews = appiumDriver.findElementsByClassName("android.widget.TextView");
                     if (textViews.size() > 0) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)